### PR TITLE
[feature-17] Add way to detect which hexagon the player click on

### PR DESCRIPTION
**Pull request by lucas-mrq**
Detect where the user click on the map

### CODE REVIEW

File: src/client/GameWindow.cpp [hunk 1]
```diff
@@ -80,9 +80,52 @@ GameWindow::GameWindow() {
...
                 case sf::Event::MouseButtonPressed:
+            
+                    clickStartingPoint = {  sf::Mouse::getPosition(clientGameWindow).x, 
+                                            sf::Mouse::getPosition(clientGameWindow).y};
+
+                    if (!mooveMode) {
+
+                        int minimumDistance = WINDOW_LENGTH;
+                        std::array<int, 2> hexagonOnClick = {0, 0};
+                        std::array<int, 2> firstHexagon = {WINDOW_LENGTH, WINDOW_WIDTH};
```
Comment (jard-111) [inline src/client/GameWindow.cpp:91]: I don't think "firstHexagon" is self explanatory name as this variable represents screen dimensions.

File: src/client/GameWindow.cpp [hunk 2]
```diff
@@ -80,9 +80,52 @@ GameWindow::GameWindow() {
...
+
+                                if (firstHexagon[0] > spriteBounds.left) firstHexagon[0] = spriteBounds.left;
+                                if (firstHexagon[1] > spriteBounds.top) firstHexagon[1] = spriteBounds.top;
+
+                                if (spriteBounds.intersects(cursorRect)) {
+
+                                    isClickable = true;
+
+                                    int distance = pow( pow(spriteBounds.left + spriteBounds.width/2 - cursorRect.left, 2) 
+                                                    + pow(spriteBounds.top + spriteBounds.height/2 - cursorRect.top, 2), 0.5);
```
Comment (jard-111) [inline src/client/GameWindow.cpp:115]: You can use "sqrt" function.
It can also be useful to create a "calcul distance" function.

**Review by jard-111 — APPROVED** [review]

File: src/client/GameWindow.cpp (hunk 1 above)
Comment (lucas-mrq) [reply src/client/GameWindow.cpp]: Done

File: src/client/GameWindow.cpp (hunk 2 above)
Comment (lucas-mrq) [reply src/client/GameWindow.cpp]: Done

**Pull request merged**